samples/transactional-session/cosmosdb/TransactionalSession_1/Backend/OrderReceivedHandler.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Logging;
+using NServiceBus;
+
+class OrderReceivedHandler : IHandleMessages<OrderReceived>
+{
+    readonly ILogger<OrderReceivedHandler> logger;
+
+    public OrderReceivedHandler(ILogger<OrderReceivedHandler> logger)
+    {
+        this.logger = logger;
+    }
+
+    public async Task Handle(OrderReceived message, IMessageHandlerContext context)
+    {
+        logger.LogInformation($"Received {nameof(OrderReceived)} event for order #{message.OrderId}");
+
+        var session = context.SynchronizedStorageSession.CosmosPersistenceSession();
+
+        var order = await session.Container.ReadItemAsync<OrderDocument>(message.OrderId, new PartitionKey(message.CustomerId));
```

Comment:
The partition key is on the session. I think it would be better to use that because the `CustomerId` must already have been propagated